include/network/Network.h

```diff
@@ -82,8 +83,23 @@ template <typename T> class Network : public NetworkBase<T> {
     sycl::queue &get_queue() { return m_q; }
     const sycl::queue &get_queue() const { return m_q; }
 
-    virtual void set_weights_matrices(const std::vector<T> &weights) {
-        m_weights_matrices.copy_from_host(weights).wait();
+    virtual void set_weights_matrices(const std::vector<T> &weights, bool weights_are_packed) {
```

Comment:
please write a comment what "weights_are_packed" does (does it mean the weights coming in are packed or the weights of the class are assumed to be packed). 

const bool